pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -0,0 +1,481 @@
+                    (node_name, time_slice)
+                    Here, node_name can be a hashable python object
+                    while the time_slice is an integer value, which
+                    denotes the index of the time_slice that the node
+                    belongs to.
+
+        EXAMPLE
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['d', 'i'])
+        >>> dbn.add_edge((('d',0), ('i',0)))
+        """
+        edge = (start, end)
+        if not any(map(lambda x: isinstance(x, tuple), iter(edge))):
+            raise AttributeError('The edge is not of a standard form')
+
+        if not any(map(lambda x: isinstance(x[1], int), iter(edge))):
+            raise AttributeError('The edge does not contain the timeslice in integer form')
+
+        if not any(map(lambda x: x[0] in self.list_of_nodes, iter(edge))):
+            new_nodes = list(filter(lambda x: x[0] not in self.list_of_nodes, iter(edge)))
+            self.add_nodes_from(set(map(lambda x: x[0], iter(new_nodes))))
+
+        if edge[0][1] == edge[1][1]:
+            if edge[0][1] in (0, 1):
+                new_edge = map(lambda x: (x[0], 1 - edge[0][1]), edge)
+                super().add_edge(*edge_type(*new_edge))
+                self.intra_edges.extend([edge_type(*edge), edge_type(*map(lambda x: (x[0], 1 - edge[0][1]), edge))])
+            else:
+                self.intra_edges.extend([edge_type(*edge)])
+        else:
+            self.inter_edges.append(edge_type(*edge))
+
+        super().add_edge(*edge_type(*edge))
+
+    def get_intra_slice(self, timeslice):
+        """
+        returns the intra slice edges present in a single time slice.
+
+        Parameters:
+        ----------
+        timeslice: integer value. should be ranging from 0
+                   to the number of the slices present in
+                   the bayesian network.
+
+        EXAMPLE
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['D','G','I','S','L'])
+        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('G',0),('L',0))])
+        >>> dbn.intra_slice(0)
+        [edge_type(start=('D', 0), end=('G', 0)),
+        edge_type(start=('I', 0), end=('G', 0)),
+        edge_type(start=('G', 0), end=('L', 0))]
+        """
+        if not isinstance(timeslice, int):
+            raise ValueError('the timeslice is not an integer value')
+        if timeslice < 0:
+            raise ValueError('negative values for a timeslice are not permissible')
+        if timeslice > self.timestate:
+            raise ValueError('the bayesian network does not contain the intra slice edges of this timeslice')
+        return list(filter(lambda x: x.start[1] == x.end[1] == timeslice, self.intra_edges))
+
+    def get_inter_slice(self, timeslice):
```

Comment:
@palashahuja Why do you need `timeslice`? The edges would be same in all the timeslices except for the first.